### PR TITLE
doc: Add `near-network` reference level documentation `Routing table computation`

### PR DESCRIPTION
**Pull request by pmnoxx**
This is part of NetworkDocumentation.md it covers:

-  Routing table computation

See https://github.com/near/nearcore/issues/5156

### CODE REVIEW

File: docs/network.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# 8 Routing table computation
+
+Routing table computation does a few things:
+- for each peer `B`, calculates set of peers `|C_b|`, such that each peer is on the shortest path to `B`.
```
Comment (bowenwang1996) [inline docs/network.md:4]:
```suggestion
- for each peer `B`, calculates set of peers `|C_b|`, such that each peer is on a shortest path to `B`.
```

Comment (bowenwang1996) [reply docs/network.md:4]: Also worth documenting how "distance" is measured when talking about shortest paths.

Comment (pmnoxx) [reply docs/network.md]: @bowenwang1996 Thanks, I added the following
```
- The distance is calculated as the minimum number of nodes on the path from given node `A`, to each other node 
on the network. That is, `A` has a distance of `0` to itself. It's neighbors will have a distance of `1`.
The neighbors of theirs neighbors will have a distance of `2`, etc.
```

File: docs/network.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+## 8.3 Step 3
+`RoutingTableActor` sends `RoutingTableUpdateResponse` message back to `PeerManagerActor`.
+
+`PeerManagerActor` keep local copy of `edges_info`, called `local_edges_info` containing only edges adjacent to current node.
+
+- This message contains list of local edges, which `PeerManagerActor` should remove.
+- `peer_forwarding` which represent on how to route messages in the P2P network
+- `peers_to_ban` - list of peers to ban for sending us edges, which failed validation in `EdgeVerifierActor`.
+
+```rust,ignore
```
Comment (bowenwang1996) [inline docs/network.md:41]: Same here

File: docs/network.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+- The distance is calculated as the minimum number of nodes on the path from given node `A`, to each other node 
+on the network. That is, `A` has a distance of `0` to itself. It's neighbors will have a distance of `1`.
+The neighbors of theirs neighbors will have a distance of `2`, etc.
+
+## 8.1 Step 1
+`PeerManagerActor` runs a `update_routing_table_trigger` every `UPDATE_ROUTING_TABLE_INTERVAL` seconds.
+
+`RoutingTableMessages::RoutingTableUpdate` message is sent to `RoutingTableActor` to request routing table re-computation.
+
+```rust,ignore
```
Comment (bowenwang1996) [inline docs/network.md:15]: why `ignore`?

**Review by bowenwang1996 — APPROVED** [review]

**Pull request merged**